### PR TITLE
feat(manage): add more fields to the GameResource form

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds a few more fields to the `GameResource` edit form in the management app. The added fields are:

* `released_at`
* `released_at_granularity`
* `Developer`
* `Publisher`
* `Genre`

Screenshot 2024-06-20 at 5 38 26 PM

Developer, Publisher, and Genre aren't yet connected to hubs-related stuff. Similar to the game page's devbox, these fields accept plain-text values.

### CODE REVIEW

File: app/Filament/Resources/GameResource.php [hunk 1]
```diff
@@ -189,6 +201,57 @@ public static function form(Form $form): Form
...
+                Forms\Components\Section::make('Earliest Release Date')
+                    ->icon('heroicon-c-calendar-days')
+                    ->description("
+                        Provide the game's earliest known release date to improve searching, sorting, and filtering on the site.
+                        Use the Precision control to specify if the release day or month are unknown.
+                    ")
+                    ->columns(['md' => 2, 'xl' => 3, '2xl' => 4])
+                    ->schema([
+                        Forms\Components\Placeholder::make('released_at_display')
+                            ->label('Preview')
```
Comment (wescopeland) [inline app/Filament/Resources/GameResource.php:214]: Originally, I wanted to dynamically adjust the datepicker's `displayFormat()` based on the user's selected granularity choice.

However, it seems the datepicker control doesn't support this functionality. The complexity of building a custom control for this use case is relatively high. This seems like the happy medium.

Comment (Jamiras) [reply app/Filament/Resources/GameResource.php:234]: Should the display (non-edit page) show the granular (preview) value?
image

**Review by Jamiras — APPROVED** [review]

File: app/Filament/Resources/GameResource.php (hunk 1 above)
Comment (Jamiras) [reply app/Filament/Resources/GameResource.php:234]: Release Date and Release Date Precision don't appear in the audit log.

Comment (wescopeland) [reply app/Filament/Resources/GameResource.php:252]: Good points - should both be fixed in latest.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**